packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -3291,76 +3336,44 @@ namespace Ifpack2 {
       // being checked, this function immediately returns false. If a check must
       // be done at this iteration, it waits for the reduction triggered by
       // ireduce to complete, then checks the global norm against the tolerance.
-      bool checkDone (const int& sweep, const magnitude_type& tol2, const bool force = false) {
+      bool checkDone (const int sweep, const magnitude_type tol2, const bool force = false) {
         // early return 
-        if (sweep <= 0) return false;
-
-#ifdef HAVE_IFPACK2_BLOCKTRIDICONTAINER_TIMERS
-        TEUCHOS_FUNC_TIME_MONITOR("BlockTriDi::NormManager::CheckDone");
-#endif
+        if (sweep <= 0) return false;	
+	
+	IFPACK2_BLOCKTRIDICONTAINER_TIMER("BlockTriDi::NormManager::CheckDone");
+	
         TEUCHOS_ASSERT(sweep >= 1);
         if ( ! force && (sweep - 1) % sweep_step_) return false;
         if (collective_) {
 #ifdef HAVE_IFPACK2_MPI
-# if MPI_VERSION >= 3
+# if defined(IFPACK2_BLOCKTRIDICONTAINER_USE_MPI_3) 
```

Comment:
Is there a global flag for MPI_3 that we can use ? Defining one for each solver will make our already unmanageable cmake scripts worse

Reply:
It would be better to use `Tpetra_Details_iallreduce.hpp` here; it has a fall-back path (that's still nonblocking) for the `MPI_VERSION < 3` case.

We don't actually test Trilinos with OpenMPI < 1.10, so perhaps we don't have to worry about this until someone complains.

Reply:
this is not a cmake flag. the macro is just replacement of MPI_VERSION >= 3. see the beginning of the code. i do this way because I want to separate the effect of mpi 3 under my control. previously, it just use mpi 3 as it uses mpi macro.

Reply:
Strange, but ok ...